### ISSUE TITLE
Button - kind and disabled button doesn't apply when hover and plain

### ISSUE THREAD

**Issue by ShimiSun** [grommet/grommet#4492]
### Expected Behavior
`disabled` styles should apply when `hoverindicator` or `plain` is set on a grommet button.

### Actual Behavior
No disabled styles applied.

### URL, screen shot, or Codepen exhibiting the issue
Use this code when using the kind button, and see how disabled styles don't apply.
```        
<Button
          label="Disabled"
          disabled
          hoverIndicator
          plain
          icon={<FormNext />}
        />
```
Related to https://github.com/grommet/grommet-theme-hpe/pull/110

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: 2.15.0
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by abnersajr** [comment]
@ShimiSun This PR can be closed. 🎉

**Comment by ShimiSun** [comment]
Thanks @abnersajr ❤️ I think there is still an issue with how `plain` Button (kind) doesn't apply disabled properly, would you like to tackle it?

**Comment by leossantiago** [comment]
@ShimiSun did a quick overview and it looks like plain Button is applying disabled properly aswell.

**Comment by ShimiSun** [comment]
@leonardodino thanks buddy, from what I've seen, Button + plain + disabled + kind (kind is when button.default is defined on the theme), the plain button doesn't get the text opacity or the `text-weak` property, so the Button `disabled` text isn't reflecting the disabled property well.

**Comment by taysea** [comment]
After investigating this before, I've come to a couple conclusions.
1. When a Button is `plain` and `disabled`, the only disabled styling comes from [this line](https://github.com/grommet/grommet/blob/master/src/js/components/Button/StyledButtonKind.js#L183):
```
${props => props.disabled && disabledStyle(props.theme.button.disabled.opacity)}
```
where `disabledStyle` only controls `opacity`.
2. When a theme defines disabled opacity of `1.0` via `props.theme.button.disabled.opacity`, then the disabled, plain button will have an opacity of 1.0 (meaning that it will appear that same as it does when it's enabled. 
3. For themes that defined disabled opacity as 1.0, they are typically using other means to indicate disabled state such as shifting the text color from `text` to `text-weak`. However, this shift in styling does not get applied to the plain buttons.

Approaches to fix:
1. Allow `plain` to be added to button theming with an opacity option (`props.theme.button.disabled.plain.opacity`) where this would override the general button disabled opacity.
2. Allow `plain` Button to also define disabled text color via theme `props.theme.button.disabled.plain.color`. However, I would need to explore this because we can't always anticipate what kind of children may be placed inside a `plain Button` and we would need to see if the text color trickles down properly.

**Comment by leossantiago** [comment]
Hi everyone! I've been also giving a shot on this issue in the past couple weeks, but lately, I've been a little busy and didn't have time to work on it.
Since @taysea pointed out some things, and @ShimiSun opened a PR addressing the issue, I'll try to contribute with my thoughts on that.

I did some testing using four buttons:
1. Button + disabled
2. Button + disabled + plain
3. Button + disabled + plain + hoverIndicator
4. Button + disabled + hoverIndicator

Since `hoverIndicator` doesn't apply in disabled buttons, 3 and 4 are discarted. Same for Kind buttons.

Problems noticed:
1. Normal Buttons and Kind buttons are not getting the proper color when disabled
2. (Normal) Icon color doesn't follow any property (like disabled color). It only changes when color is passed as a prop of the element passed as icon prop. Kind follows disabled color (and also color prop) normally.
3. (Kind) plain buttons don't receive text-weak color. Non-plain buttons receive that properly.

Kind non-plain receives styling from theme.button.disabled and disabledStyle(props.theme.button.disabled.opacity), while kind plain only receives styling from disabledStyle.

Like taysea, I've been thinking about ways of fixing kind+plain, but since #4589 deprecates it, I think there's no need anymore to fix that.

I'd still suggest:
1. Fix buttons text color when disabled (they all have opacity according to theme). That's achiavable by defining a 'text-weak' alongside 'opacity' in theme. Perharps it could be done in the PR Shimi already opened, maybe?
2. Make icon color follow disabled color. But I'd like to ask if we should worry about that, since is stated in Button.js that normal buttons will probably be deprecated in the future.

Let me know what u guys think :)

**TL;DR:** PR #4589 deprecates kind+plain buttons, which solves part of the problem, but all buttons are not getting the proper disabled color, and I think we should fix that. Also, we might make button Icon color (non kind) to follow disabled color.

**Comment by ShimiSun** [comment]
I think we have implemented all possible action items for this issue as linked to the PRs above.
With button kind, users can define disabled text color, border color, etc but plain will only look at the disabled opacity value `props.theme.button.disabled.opacity`.
The issue we experienced with the HPE theme, is that the opacity is set to 1.0 so you don’t see any drop in opacity on disabled + kind + plain Button. But, if any other theme relied on the grommet default opacity (0.3) or defined their own (as long as it’s not 1.0) then the disabled opacity would give a visual effect of “disabled”. In the case of HPE theme users, it will be recommended not to use `plain` buttons.

**Comment by ShimiSun** [comment]
Linked https://github.com/grommet/grommet-theme-hpe/issues/93
